Source/core/CyclicBuffer.h

```diff
@@ -252,7 +252,7 @@ namespace Core {
         void AdminLock();
         void AdminUnlock();
         void Reevaluate();
-        uint32_t SignalLock(const uint32_t waitTime);
+        uint32_t SignalLock(uint32_t& waitTime);
```

Comment:
What is the reason for changing this interface ? Why do we need to return the amount of time we waited ?